app/api/entities/entities.js

```diff
@@ -347,6 +347,16 @@ const withDocuments = (entities, documentsFullText, withPdfInfo) =>
     })
   );
 
+const reindexEntitiesByTemplate = async (template, options) => {
+  const templateHasRelationShipProperty = !template.properties?.find(
+    p => p.type === propertyTypes.relationship
+  );
+  if (options.reindex && (options.generatedIdAdded || templateHasRelationShipProperty)) {
```

Comment:
Assuming that templateHasRelationShipProperty is postponing the reindex. If that's the case then should this be postponed even if generatedIdAdded is true?

Reply:
Not really, the postponed reindex don't do a reindex of all the entities, just those entities that were impacted by the changes, for example, the entities without a relationship property, are not reindexed, this is why although it implies a double indexation it's necessary to do it here in order to have all the generated ids indexed.